File: practica con flask/modulos/zoo_animals.py
import json
import random

class zoo_animals:

    def __init__(self):
        self.file = "data/animales.json"

    def obtener_animal(self):
        try:
            with open(self.file, "r", encoding="utf-8") as f:
                animales = json.load(f)

            animal = random.choice(animales)

            return {
                "name": animal.get("name"),
                "animal_type": animal.get("animal_type"),
                "active_time": animal.get("active_time"),
                "lifespan": animal.get("lifespan"),
                "habitat": animal.get("habitat"),
                "diet": animal.get("diet"),
                "geo_range": animal.get("geo_range"),
                "image_link": animal.get("image_link")
            }

        except Exception as e:
            return {"error": f"Error al obtener el animal: {e}"}

File: practica con flask/static/js/zoo_animals.js
const animalImage = document.getElementById('zoo_animal_image');
const animalName = document.getElementById('zoo_animal_name');
const animalHabitat = document.getElementById('zoo_animal_habitat');
const animalDiet = document.getElementById('zoo_animal_diet');
const animalLifespan = document.getElementById('zoo_animal_lifespan');
const animalActiveTime = document.getElementById('zoo_animal_active_time');
const animalGeoRange = document.getElementById('zoo_animal_geo_range');
const animalType = document.getElementById('zoo_animal_type');

// ===== FUNCIÓN PARA OBTENER Datos =====
// Hace fetch a la API de Flask y actualiza los datos

async function obtenerZooAnimal() {
    try {
        // Hacer la solicitud a la API
        const response = await fetch("/animales/api/zoo/");
        // Verificar si la respuesta es exitosa
        if (!response.ok) {
            throw new Error(`Error: ${response.status}`);
        }

        // Convertir respuesta a JSON
        const datos = await response.json();

        // Actualizar los elementos del DOM con los datos obtenidos
        animalImage.src = datos.image_link;
        animalName.textContent = datos.name;
        animalHabitat.textContent = datos.habitat;
        animalDiet.textContent = datos.diet;
        animalLifespan.textContent = datos.lifespan;
        animalActiveTime.textContent = datos.active_time;
        animalGeoRange.textContent = datos.geo_range;
        animalType.textContent = datos.animal_type;

        console.log('✓ Información del animal actualizada correctamente');
    } catch (error) {
        console.error('✗ Error al obtener información del animal:', error);
    }
}

// ===== LLAMADAS A LA FUNCIÓN =====
// Obtener información del animal al cargar la página
obtenerZooAnimal();

// Obtener nueva información del animal cada 15 segundos (15000 milisegundos)
setInterval(obtenerZooAnimal, 15000);